firebase-common/src/test/java/com/google/firebase/heartbeatinfo/DefaultHeartBeatControllerTest.java

```diff
@@ -202,27 +194,32 @@ public void generateHeartBeat_twoHeartBeatsSameUserAgent()
     heartBeatController
         .getHeartBeatsHeader()
         .addOnCompleteListener(executor, getOnCompleteListener);
-    JSONObject output = new JSONObject();
-    JSONArray array = new JSONArray();
-    JSONObject obj = new JSONObject();
-    obj.put("agent", "test-agent");
-    obj.put("date", dateList);
-    array.put(obj);
-    output.put("heartbeats", array);
-    output.put("version", "2");
-    String expected =
-        Base64.getUrlEncoder()
-            .withoutPadding()
-            .encodeToString(compress(output.toString()).getBytes());
+    String str =
+        "{\"heartbeats\":[{\"agent\":\"test-agent\",\"dates\":[\"2015-03-02\",\"2015-03-01\"]}],\"version\":\"2\"}";
+    String expected = compress(str);
     assertThat(getOnCompleteListener.await().replace("\n", "")).isEqualTo(expected);
   }
 
+  private static String base64Encode(byte[] input) {
+    return Base64.getUrlEncoder().withoutPadding().encodeToString(input);
+  }
+
+  private static byte[] gzip(String input) {
+    ByteArrayOutputStream byteOutputStream = new ByteArrayOutputStream();
+    try {
+      try (GZIPOutputStream gzipOutputStream = new GZIPOutputStream(byteOutputStream)) {
+        gzipOutputStream.write(input.getBytes(UTF_8));
+      }
+      byte[] gzipped = byteOutputStream.toByteArray();
+      byteOutputStream.close();
```

Comment:
Looks like you can use try with resources here as well

Reply:
it complains saying byte[] is not a proper type for try with resources.

Reply:
AFAIK not closing it is OK.

https://docs.oracle.com/javase/7/docs/api/java/io/ByteArrayOutputStream.html#close()

> Closing a ByteArrayOutputStream has no effect. The methods in this class can be called after the stream has been closed without generating an IOException.